modules/light-clients/08-wasm/types/msgs.go

```diff
@@ -36,3 +36,35 @@ func (m MsgStoreCode) ValidateBasic() error {
 
 	return nil
 }
+
+// MsgMigrateContract creates a new MsgMigrateContract instance
+func NewMsgMigrateContract(signer string, clientID string, codeHash []byte, migrateMsg []byte) *MsgMigrateContract {
+	return &MsgMigrateContract{
+		Signer:   signer,
+		ClientId: clientID,
+		CodeHash: codeHash,
+		Msg:      migrateMsg,
+	}
+}
+
+// ValidateBasic implements sdk.Msg
+func (m MsgMigrateContract) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(m.Signer)
+	if err != nil {
+		return errorsmod.Wrapf(ibcerrors.ErrInvalidAddress, "string could not be parsed as address: %v", err)
+	}
+
+	if err := ValidateWasmCodeHash(m.CodeHash); err != nil {
+		return err
+	}
+
+	if err := ValidateClientID(m.ClientId); err != nil {
+		return err
+	}
+
+	if len(m.Msg) == 0 {
+		return errorsmod.Wrap(ibcerrors.ErrInvalidRequest, "migrate message cannot be empty")
+	}
+
+	return nil
```

Comment:
could also see a cheeky refactor like this

```suggestion
	_, err := sdk.AccAddressFromBech32(m.Signer)
	if err != nil {
		return errorsmod.Wrapf(ibcerrors.ErrInvalidAddress, "string could not be parsed as address: %v", err)
	}

	if len(m.Msg) == 0 {
		return errorsmod.Wrap(ibcerrors.ErrInvalidRequest, "migrate message cannot be empty")
	}

	if err := ValidateWasmCodeHash(m.CodeHash); err != nil {
		return err
	}

        return ValidateClientID(m.ClientId)
```

Reply:
Yeah I noticed this too. I remember at some point, @damiannolan said he prefers explicit nil returns in a different context.

Reply:
fair enough, I'd tend to agree with larger funcs where I wouldn't be able to see sig